force-app/main/default/classes/PMT_RefundController.cls

```diff
@@ -44,34 +44,91 @@ public with sharing class PMT_RefundController {
         }
         set;
     }
+ 
+    @TestVisible
+    private static PMT_RefundService refundService {
+        get {
+            if (refundService == null) {
+                refundService = new PMT_RefundService();
+            }
+            return refundService;
+        }
+        set;
+    }
 
     @AuraEnabled
-    public static RefundView getPermissionData() {
+    public static RefundView getInitialView(Id paymentId) {
         RefundView refundView = new RefundView();
-        refundView.hasRequiredPermissions = hasRequiredPermission();
+        setPermissionData(refundView);
 
+        if (!refundView.hasRequiredPermissions) {
+            return refundView;
+        }
+
+        npe01__OppPayment__c originalPayment = paymentSelector.getPaymentRecordForRefund(paymentId);
```

Comment:
Does this query need to happen if we're viewing the paid Payment that should be refunded? There will be no "original" payment in that case.

Reply:
I guess it's the variable name that causes the confusion. This query is to query the original payment's Amount, Paid Date, and IsoCurrencyCode, which will be displayed on the modal. I am reusing the same query as the other refund query because we are using a subset of the fields that will be needed for processing a refund.